CorsixTH/CMakeLists.txt

```diff
@@ -110,13 +104,21 @@ find_package(Lua REQUIRED)
 if(Lua_FOUND)
   target_link_libraries(CorsixTH ${LUA_LIBRARY})
   include_directories(${LUA_INCLUDE_DIR})
-  # Special link flags needed on OSX/64bit, according to: http://luajit.org/install.html
+  # Special link flags needed on macOS, according to: http://luajit.org/install.html
   # If these are not specified, luaL_newstate() returns NULL and we get this:
   #   Fatal error starting CorsixTH: Cannot open Lua state.
-  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND LUA_INTERPRETER_TYPE STREQUAL "LuaJIT" AND CMAKE_SIZEOF_VOID_P EQUAL 8)
+  if(APPLE AND LUA_LIBRARIES MATCHES "libluajit")
```

Comment:
Should have been called out that you're dropping compiling for 32bit macOS (I know Catalina drops it, but not everyone is on Catalina).

Alternately leave in the `CMAKE_SIZEOF_VOID_P` condition.

Reply:
10.6, Snow Leopard, was the last to support 32bit hardware. I can leave it in, for people who intentionally built 32bit lua?

Reply:
Might as well.